src/components/GptSearchBar.tsx

```diff
@@ -0,0 +1,87 @@
+import { useDispatch, useSelector } from "react-redux";
+import lang from "../utils/languageConstant";
+import { RootState } from "../utils/appStore";
+import { useRef } from "react";
+import model from "../utils/geminiai";
+import { API_OPTIONS } from "../utils/constants";
+import { addGptMovies, addGptSearchedMovies } from "../utils/gptSlice";
+
+const GptSearchBar = () => {
+  const currentLanguage = useSelector((store: RootState) => store.config?.lang);
+  const searchInputText = useRef(null);
+  const dispatch = useDispatch();
+
+  const handleSearchBarSumbit = (e: React.ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+  };
+
+  const searchMovieTMDB = async (movie: string) => {
+    const data = await fetch(
+      "https://api.themoviedb.org/3/search/movie?query=" +
+        movie +
+        "&include_adult=false&language=en-US&page=1",
+      API_OPTIONS
+    );
+    const jsonData = await data.json();
+
+    return jsonData.results;
+  };
+
+  const handleGotSearchClick = async () => {
```

Comment:
It's an event click, pretty sure adding an async here wouldn't make much difference

Reply:
I am using await keyword inside